### DISCUSSION TITLE
write flatten n-dimensional array

### DISCUSSION THREAD

**Discussion by aghaeifar** [BlueBrain/HighFive#981] [Questions] [answered]
Hi all

I store a flattened n-dimensional array within a 1D vector. Is it feasible to write this 1D vector as an n-dimensional array to an HDF5 file? I am hesitant to create a new n-dimensional array and duplicate the data, as this approach is not memory-efficient. when I attempt something like:

```
File file("create_dataset_example.h5", File::Truncate);
std::vector<size_t> dims{30, 100000, 1, 1};
DataSet dataset = file.createDataSet<float>("dset", DataSpace(dims));
dataset.write(my_1D_vector);
```

the program throws an exception:
`what():  Impossible to write buffer of dimensions [ 30, 100000, 1, 1] into dataset with n = 1 dimensions.
`
Can anyone advice? 
Thanks.

**Comment by aghaeifar** [comment, marked as answer]
Apparently there is a good example for this:
https://github.com/BlueBrain/HighFive/blob/master/src/examples/read_write_raw_ptr.cpp

**Reply by CsatiZoltan** [reply]
Thanks, it helped me with the same problem that you had!
